### ISSUE TITLE
Align theme to figma planning

### ISSUE THREAD

**Issue by bhoppers2008** [jpmorganchase/salt-ds#4486]
The new figma libraries release (end of 2025) simplifies our figma libraries down to a reduced set, primarily built around 1 set of components supporting 2 themes–UITK Legacy and JPM Brand. 

How do we align the theme in code to mirror what design teams will providing tech teams in their documentation?

**Notes from @bhoppers2008 discussion with @origami-z (10 Dec)**

- Salt will provide two themes (uitk and brand) for technical components. 
- Changing from these defaults is something a Business Library will own (i.e., i want the brand theme but blue buttons... not a Salt recommendation–technically possible but teams will have to maintain it themselves).
- How do we get ourselves to a state where the defaults in code and figma are the same?
    - Salt Provider – Stays the same
    - Salt Provider Next – Deprecate
        - Or add a prop to the Next which changes the defaults
        - We still wouldn’t have the ‘Next Legacy’ in code though
    - Salt Provider Brand – Updated to be the same as the brand in Figma
        - Capability to support a Next Legacy theme but not the Salt recommendation–needs to be owned by the Business Library
        - Are there any open-source problems if we explicitly name the theme ‘JPM Brand’ and set the defaults to brand? Better as something more generic in code (i.e., ‘Brand’). Check with Sean if this is ok?
 
- Should we consider removing ‘Accent’ from the new provider? Other styling options remain relevant under the hood but ‘accent’ is the difference between the UITK Legacy (blue) and JPM Brand (teal).

**Comment by joshwooding** [comment]
Have we considered just adding support the the current SaltProvider? We have a `theme` prop that could be set to "brand". This would potentially be more future proof as we'd avoid adding another provider and it might work if they are using SaltProviderNext already.

**Comment by mark-tate** [comment]
@bhoppers2008 can we chop this up into smaller deliverables?

**Comment by mark-tate** [comment]
Cayenne Goal: meeting 
In the next 3 weeks turn the unknowns into knows and form a plan/message for Q2

**Comment by origami-z** [comment]
```js 
// Salt exports
export const brandOptions = { accent: "teal", corner: "rounded", headingFont: "Amplitude", actionFont: "Amplitude" };
```
Then user could do `<SaltProviderNext {...brandOptions} />`

**Comment by mark-tate** [comment]
Cayenne Goal:  how to apply brand options and what message we give to consumers.

**Comment by mark-tate** [comment]
Cardamon Goal: discussion required: how to apply brand options and what message we give to consumers.

**Comment by origami-z** [comment]
Met on Monday, Ben updated theme roadmap on https://go/salt-ds-internal/product/roadmap/themes/index#updating-the-salt-provider-components

Plan of action for code, all of below needs carefully testing to cater for all existing users, so would likely take more than one sprint to implement and verify: 

- **Biggest challenge**:  We need to include new CSS rules (currently in `packages/theme/css/theme-next.css`) into the main `index.css` to make all below to work.
    - This change (addition to existing `@salt-ds/theme/index.css`) should be verified first, before all other changes to be made.
    - Currently we use `.salt-theme-next` class selector, which we should consider move away and use data-* attribute to denote theme choice just like other styling options. 
    - One potential solution: ignore all `*-next.css`, and modify existing files, with new attribute matcher. If we do this, need to verify existing next theme users would not be impacted by the change of theme scope specificity. 
- Add prop(s) to `SaltProvider` to toggle between 2 supported themes. 
    - To maintain backwards compatibility, need a different prop name than the current `theme`, maybe something like `builtInTheme?: "legacy" | "brand"` (name is **not** finalized as of 17 Apr). Defaults to `"legacy"`.
    - When `"legacy"`, only `mode` and `density` would be taking effect.
    - When `"brand"`, all styling options will be taken effect, thus prop type needs to be adjusted. Add `JSDocs` comments to make sure it is clear to the user
        - Default values of the new styling options should reflect brand choices, which is different from `SaltProviderNext` default at the moment
- `SaltProviderNext` - add a note about preference of new prop in `SaltProvider`, and difference in default props. Mention its future plan of removal in favor of above. 
    - Do not mark it with `@deprecated` now. We don't want to send panic signal to all existing users on it.
    - Migration notes should be added, especially around default prop value differences
- Test nested providers use cases and its inheritance correctness
    - Different prop values of `SaltProvider`
    - Mix `SaltProvider` and `SaltProviderNext`
    - Auto-detection in several Salt internal implementation, e.g. `DefaultFloatingComponent` in `packages/core/src/utils/useFloatingUI/useFloatingUI.tsx`, update implementation
- `useTheme` hook needs some touch as well
    - `builtInTheme` should be added to the hook return
    - Existing `themeNext` should be left as it is to denote `SaltProvider` / `SaltProviderNext`, given `buildInTheme` would mean different default setting when choosing different provider. It would be considered semi-deprecated when SaltProviderNext is removed in the future.

**Comment by libertymayc** [comment]
> One potential solution: ignore all *-next.css, and modify existing files, with new attribute matcher. If we do this, need to verify existing next theme users would not be impacted by the change of theme scope specificity.

small technical comment here, if users have added their own theme, e.g. `.studio.salt-theme-next`, and salt theme next class were removed, this would break 

> Should we consider removing ‘Accent’ from the new provider? Other styling options remain relevant under the hood but ‘accent’ is the difference between the UITK Legacy (blue) and JPM Brand (teal).

I would agree with this - since data-accent is only used at the theme level in code and also it appears inconsistent if a team is providing accent overrides, e.g. we use 'cobalt' as our accent, which is done through the theme only (no prop) ... but the prop would still be e.g. accent="blue" (as default) as shown in the html tag, which is somewhat confusing. 
TLDR 'JPM brand' but with cobalt makes more sense technically than 'blue accent' but with cobalt

**Comment by origami-z** [comment]
> small technical comment here, if users have added their own theme, e.g. `.studio.salt-theme-next`, and salt theme next class were removed, this would break

Yes, not planning to remove `.salt-theme-next` before the breaking change. We can add additional scoping to existing one, e.g. `.salt-theme[data-mode='brand']`. Effectively, in v1, we need to support both approaches. And write migration guide for v2 to transition.

**Comment by origami-z** [comment]
Follow up work created in #5018
